### PR TITLE
add files for QuickStart tutorial

### PR DESCRIPTION
**Pull request by frx-wintermute**
This pull request is part of plan (B) described in https://github.com/su2code/su2code.github.io/pull/134#issuecomment-1616012091

### CODE REVIEW

File: compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+% Physical governing equations (EULER, NAVIER_STOKES,
+%                               WAVE_EQUATION, HEAT_EQUATION, FEM_ELASTICITY,
+%                               POISSON_EQUATION)
+SOLVER= EULER
+%
+% Mathematical problem (DIRECT, CONTINUOUS_ADJOINT)
+MATH_PROBLEM= CONTINUOUS_ADJOINT
+%
+% Restart solution (NO, YES)
+RESTART_SOL= NO
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg]:
```suggestion
SOLVER= EULER
MATH_PROBLEM= CONTINUOUS_ADJOINT
RESTART_SOL= NO
```

File: compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+MACH_NUMBER= 0.8
+%
+% Angle of attack (degrees)
+AOA= 1.25
+%
+% Free-stream pressure (101325.0 N/m^2 by default, only Euler flows)
+FREESTREAM_PRESSURE= 101325.0
+%
+% Free-stream temperature (273.15 K by default)
+FREESTREAM_TEMPERATURE= 273.15
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg]:
```suggestion
MACH_NUMBER= 0.8
AOA= 1.25
FREESTREAM_PRESSURE= 101325.0
FREESTREAM_TEMPERATURE= 273.15
```

File: compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+%
+% Reference length for pitching, rolling, and yawing non-dimensional moment
+REF_LENGTH= 1.0
+%
+% Reference area for force coefficients (0 implies automatic calculation)
+REF_AREA= 1.0
+%
+% Flow non-dimensionalization (DIMENSIONAL, FREESTREAM_PRESS_EQ_ONE,
+%                              FREESTREAM_VEL_EQ_MACH, FREESTREAM_VEL_EQ_ONE)
+REF_DIMENSIONALIZATION= DIMENSIONAL
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg]:
```suggestion
GAMMA_VALUE= 1.4
GAS_CONSTANT= 287.87

% ---------------------- REFERENCE VALUE DEFINITION ---------------------------%

REF_ORIGIN_MOMENT_X = 0.25
REF_ORIGIN_MOMENT_Y = 0.00
REF_ORIGIN_MOMENT_Z = 0.00
REF_LENGTH= 1.0
REF_AREA= 1.0
REF_DIMENSIONALIZATION= DIMENSIONAL
```

File: compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+% ------------------------ SURFACES IDENTIFICATION ----------------------------%
+%
+% Marker(s) of the surface in the surface flow solution file
+MARKER_PLOTTING = ( airfoil )
+%
+% Marker(s) of the surface where the non-dimensional coefficients are evaluated.
+MARKER_MONITORING = ( airfoil )
+%
+% Marker(s) of the surface where obj. func. (design problem) will be evaluated
+MARKER_DESIGNING = ( airfoil )
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg]:
```suggestion
MARKER_EULER= ( airfoil )
MARKER_FAR= ( farfield )

% ------------------------ SURFACES IDENTIFICATION ----------------------------%

MARKER_PLOTTING = ( airfoil )
MARKER_MONITORING = ( airfoil )
MARKER_DESIGNING = ( airfoil )
```

File: compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+%
+% Courant-Friedrichs-Lewy condition of the finest grid
+CFL_NUMBER= 1e3
+%
+% Adaptive CFL number (NO, YES)
+CFL_ADAPT= NO
+%
+% Parameters of the adaptive CFL number (factor down, factor up, CFL min value,
+%                                        CFL max value )
+CFL_ADAPT_PARAM= ( 0.1, 2.0, 10.0, 1e10 )
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg]:
```suggestion
NUM_METHOD_GRAD= WEIGHTED_LEAST_SQUARES
OBJECTIVE_FUNCTION= DRAG
CFL_NUMBER= 1e3
CFL_ADAPT= NO
CFL_ADAPT_PARAM= ( 0.1, 2.0, 10.0, 1e10 )
```

File: compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+LINEAR_SOLVER= FGMRES
+%
+% Preconditioner of the Krylov linear solver (JACOBI, LINELET, LU_SGS)
+LINEAR_SOLVER_PREC= ILU
+%
+% Minimum error of the linear solver for implicit formulations
+LINEAR_SOLVER_ERROR= 1E-10
+%
+% Max number of iterations of the linear solver for the implicit formulation
+LINEAR_SOLVER_ITER= 10
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg]:
```suggestion
LINEAR_SOLVER= FGMRES
LINEAR_SOLVER_PREC= ILU
LINEAR_SOLVER_ERROR= 1E-10
LINEAR_SOLVER_ITER= 10
```

File: compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+MG_POST_SMOOTH= ( 0, 0, 0, 0 )
+%
+% Jacobi implicit smoothing of the correction
+MG_CORRECTION_SMOOTH= ( 0, 0, 0, 0 )
+%
+% Damping factor for the residual restriction
+MG_DAMP_RESTRICTION= 1.0
+%
+% Damping factor for the correction prolongation
+MG_DAMP_PROLONGATION= 1.0
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg]:
```suggestion
MGLEVEL= 3
MGCYCLE= W_CYCLE
MG_PRE_SMOOTH= ( 1, 2, 3, 3 )
MG_POST_SMOOTH= ( 0, 0, 0, 0 )
MG_CORRECTION_SMOOTH= ( 0, 0, 0, 0 )
MG_DAMP_RESTRICTION= 1.0
MG_DAMP_PROLONGATION= 1.0
```

File: compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+% -------------------- FLOW NUMERICAL METHOD DEFINITION -----------------------%
+%
+% Convective numerical method
+%
+CONV_NUM_METHOD_FLOW= JST
+%
+% 2nd and 4th order artificial dissipation coefficients
+JST_SENSOR_COEFF= ( 0.5, 0.02 )
+%
+% Time discretization (RUNGE-KUTTA_EXPLICIT, EULER_IMPLICIT, EULER_EXPLICIT)
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg]:
```suggestion
CONV_NUM_METHOD_FLOW= JST
JST_SENSOR_COEFF= ( 0.5, 0.02 )
```

File: compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg [hunk 9]
```diff
@@ -0,0 +1,327 @@
...
+
+% ---------------- ADJOINT-FLOW NUMERICAL METHOD DEFINITION -------------------%
+%
+% Convective numerical method (JST, LAX-FRIEDRICH, ROE)
+CONV_NUM_METHOD_ADJFLOW= JST
+%
+% Reduction factor of the CFL coefficient in the adjoint problem
+CFL_REDUCTION_ADJFLOW= 0.01
+%
+% Time discretization (RUNGE-KUTTA_EXPLICIT, EULER_IMPLICIT)
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/adj/inv_adj_NACA0012.cfg]:
```suggestion
CONV_NUM_METHOD_ADJFLOW= JST
CFL_REDUCTION_ADJFLOW= 0.01
```

File: compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg [hunk 10]
```diff
@@ -0,0 +1,327 @@
...
+% Physical governing equations (EULER, NAVIER_STOKES,
+%                               WAVE_EQUATION, HEAT_EQUATION, FEM_ELASTICITY,
+%                               POISSON_EQUATION)
+SOLVER= EULER
+%
+% Mathematical problem (DIRECT, CONTINUOUS_ADJOINT)
+MATH_PROBLEM= DISCRETE_ADJOINT
+%
+% Restart solution (NO, YES)
+RESTART_SOL= NO
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg]:
```suggestion
SOLVER= EULER
MATH_PROBLEM= DISCRETE_ADJOINT
RESTART_SOL= NO
```

File: compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg [hunk 11]
```diff
@@ -0,0 +1,327 @@
...
+MACH_NUMBER= 0.8
+%
+% Angle of attack (degrees)
+AOA= 1.25
+%
+% Free-stream pressure (101325.0 N/m^2 by default, only Euler flows)
+FREESTREAM_PRESSURE= 101325.0
+%
+% Free-stream temperature (273.15 K by default)
+FREESTREAM_TEMPERATURE= 273.15
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg]:
```suggestion
MACH_NUMBER= 0.8
AOA= 1.25
FREESTREAM_PRESSURE= 101325.0
FREESTREAM_TEMPERATURE= 273.15
```

File: compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg [hunk 12]
```diff
@@ -0,0 +1,327 @@
...
+
+% -------------- COMPRESSIBLE AND INCOMPRESSIBLE FLUID CONSTANTS --------------%
+%
+% Ratio of specific heats (1.4 (air), only for compressible flows)
+GAMMA_VALUE= 1.4
+%
+% Specific gas constant (287.87 J/kg*K (air), only for compressible flows)
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg]:
```suggestion
GAMMA_VALUE= 1.4
```

File: compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg [hunk 13]
```diff
@@ -0,0 +1,327 @@
...
+REF_ORIGIN_MOMENT_Z = 0.00
+%
+% Reference length for pitching, rolling, and yawing non-dimensional moment
+REF_LENGTH= 1.0
+%
+% Reference area for force coefficients (0 implies automatic calculation)
+REF_AREA= 1.0
+%
+% Flow non-dimensionalization (DIMENSIONAL, FREESTREAM_PRESS_EQ_ONE,
+%                              FREESTREAM_VEL_EQ_MACH, FREESTREAM_VEL_EQ_ONE)
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg]:
```suggestion
REF_ORIGIN_MOMENT_X = 0.25
REF_ORIGIN_MOMENT_Y = 0.00
REF_ORIGIN_MOMENT_Z = 0.00
REF_LENGTH= 1.0
REF_AREA= 1.0
```

File: compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg [hunk 14]
```diff
@@ -0,0 +1,327 @@
...
+REF_DIMENSIONALIZATION= DIMENSIONAL
+
+% ----------------------- BOUNDARY CONDITION DEFINITION -----------------------%
+%
+% Marker of the Euler boundary (NONE = no marker)
+MARKER_EULER= ( airfoil )
+%
+% Marker of the far field (NONE = no marker)
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg]:
```suggestion
MARKER_EULER= ( airfoil )
```

File: compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg [hunk 15]
```diff
@@ -0,0 +1,327 @@
...
+
+% ------------------------ SURFACES IDENTIFICATION ----------------------------%
+%
+% Marker(s) of the surface in the surface flow solution file
+MARKER_PLOTTING = ( airfoil )
+%
+% Marker(s) of the surface where the non-dimensional coefficients are evaluated.
+MARKER_MONITORING = ( airfoil )
+%
+% Marker(s) of the surface where obj. func. (design problem) will be evaluated
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg]:
```suggestion
MARKER_PLOTTING = ( airfoil )
MARKER_MONITORING = ( airfoil )
```

File: compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg [hunk 16]
```diff
@@ -0,0 +1,327 @@
...
+CFL_NUMBER= 1e3
+%
+% Adaptive CFL number (NO, YES)
+CFL_ADAPT= NO
+%
+% Parameters of the adaptive CFL number (factor down, factor up, CFL min value,
+%                                        CFL max value )
+CFL_ADAPT_PARAM= ( 0.1, 2.0, 10.0, 1e10 )
+%
+% Number of total iterations
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg]:
```suggestion
NUM_METHOD_GRAD= WEIGHTED_LEAST_SQUARES
OBJECTIVE_FUNCTION= DRAG
CFL_NUMBER= 1e3
CFL_ADAPT= NO
CFL_ADAPT_PARAM= ( 0.1, 2.0, 10.0, 1e10 )
```

File: compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg [hunk 17]
```diff
@@ -0,0 +1,327 @@
...
+% Linear solver for implicit formulations (BCGSTAB, FGMRES)
+LINEAR_SOLVER= FGMRES
+%
+% Preconditioner of the Krylov linear solver (JACOBI, LINELET, LU_SGS)
+LINEAR_SOLVER_PREC= ILU
+%
+% Minimum error of the linear solver for implicit formulations
+LINEAR_SOLVER_ERROR= 1E-10
+%
+% Max number of iterations of the linear solver for the implicit formulation
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg]:
```suggestion
LINEAR_SOLVER= FGMRES
LINEAR_SOLVER_PREC= ILU
LINEAR_SOLVER_ERROR= 1E-10
```

File: compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg [hunk 18]
```diff
@@ -0,0 +1,327 @@
...
+% Multi-Grid PostSmoothing Level
+MG_POST_SMOOTH= ( 0, 0, 0, 0 )
+%
+% Jacobi implicit smoothing of the correction
+MG_CORRECTION_SMOOTH= ( 0, 0, 0, 0 )
+%
+% Damping factor for the residual restriction
+MG_DAMP_RESTRICTION= 1.0
+%
+% Damping factor for the correction prolongation
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg]:
```suggestion
MGLEVEL= 3
MGCYCLE= W_CYCLE
MG_PRE_SMOOTH= ( 1, 2, 3, 3 )
MG_POST_SMOOTH= ( 0, 0, 0, 0 )
MG_CORRECTION_SMOOTH= ( 0, 0, 0, 0 )
MG_DAMP_RESTRICTION= 1.0
```

File: compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg [hunk 19]
```diff
@@ -0,0 +1,327 @@
...
+%
+% Convective numerical method
+%
+CONV_NUM_METHOD_FLOW= JST
+%
+% 2nd and 4th order artificial dissipation coefficients
+JST_SENSOR_COEFF= ( 0.5, 0.02 )
+%
+% Time discretization (RUNGE-KUTTA_EXPLICIT, EULER_IMPLICIT, EULER_EXPLICIT)
+TIME_DISCRE_FLOW= EULER_IMPLICIT
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg]:
```suggestion
CONV_NUM_METHOD_FLOW= JST
JST_SENSOR_COEFF= ( 0.5, 0.02 )
TIME_DISCRE_FLOW= EULER_IMPLICIT
```

File: compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg [hunk 20]
```diff
@@ -0,0 +1,327 @@
...
+%
+% Convective numerical method (JST, LAX-FRIEDRICH, ROE)
+CONV_NUM_METHOD_ADJFLOW= JST
+%
+% Reduction factor of the CFL coefficient in the adjoint problem
+CFL_REDUCTION_ADJFLOW= 0.01
+%
+% Time discretization (RUNGE-KUTTA_EXPLICIT, EULER_IMPLICIT)
+TIME_DISCRE_ADJFLOW= EULER_IMPLICIT
+
```
Comment (pcarruscag) [inline compressible_flow/QuickStart/dadj/inv_dadj_NACA0012.cfg]:
```suggestion
```

**Review by pcarruscag — APPROVED** [review]

**Review by pcarruscag — CHANGES_REQUESTED** [review]
I think you opened the PR but did not allow commits from maintainers, you'll have to apply the suggestions above yourself unfortunately.

**Comment by frx-wintermute** [comment]
> I think you opened the PR but did not allow commits from maintainers, you'll have to apply the suggestions above yourself unfortunately.

Yes, I confirm that I opened the pull request without allowing commits from maintainers: the rationale is that I would like to understand and check any change you possibly ask me to apply, as in this very case.

If I am not mistaken, you are asking me to strip out all the comments from the two configuration files. Why so?

Other tutorials seem to have configuration files with comments. For instance: https://github.com/su2code/Tutorials/blob/develop/incompressible_flow/Inc_Laminar_Flat_Plate/lam_flatplate.cfg

Please explain the reason behind your requested changes and I will be more than happy to apply them!

**Comment by pcarruscag** [comment]
The lists of possible options in those comments have become obsolete, we want to document options in a single place (config_template) and only note what is important for particular cases.

**Comment by pcarruscag** [comment]
For example these 2 configs are copy pasted from the quick start and only a couple lines are different.

**Comment by pcarruscag** [comment]
And since they are not used for any test I would even say it would be more valuable to simply instruct what few options need to change in the tutorial, and avoid adding these files which will become outdated (note that I did ask for those tests)

**Comment by frx-wintermute** [comment]
> The lists of possible options in those comments have become obsolete, we want to document options in a single place (config_template) and only note what is important for particular cases.

I have stripped out all the comments from the two config files.

**Review by pcarruscag — APPROVED** [review]
Thank you

**Pull request merged**